File: 02.py
def parse_policy(row):
    policy_parts = row.split(" ")
    first, second = map(int, policy_parts[0].split("-"))

    return first, second, policy_parts[1][0], policy_parts[2]


def part_1():
    counter = 0
    with open("input.txt", "r", encoding="utf-8") as f:
        for row in f:
            first, second, letter, password = parse_policy(row)
            occurance = password.count(letter)

            if occurance >= first and occurance <= second:
                counter += 1

    print(counter)


def part_2():
    counter = 0
    with open("input.txt", "r", encoding="utf-8") as f:
        for row in f:
            first, second, letter, password = parse_policy(row)

            if (password[first-1] == letter) != (password[second-1] == letter):
                counter += 1

    print(counter)


if __name__ == '__main__':
    part_1()
    part_2()

File: 2022/02.jl
function read_input()
    open("02-input.txt") do file
        total_score = 0

        lines = readlines(file)
        for l in lines
            l = split(l, " ")

            if l[2] == "X"
                if l[1] == "A"
                    total_score += 3
                elseif l[1] == "B"
                    total_score += 1
                elseif l[1] == "C"
                    total_score += 2
                end
            elseif l[2] == "Y"
                total_score += 3
                if l[1] == "A"
                    total_score += 1
                elseif l[1] == "B"
                    total_score += 2
                elseif l[1] == "C"
                    total_score += 3
                end
            elseif l[2] == "Z"
                total_score += 6
                if l[1] == "A"
                    total_score += 2
                elseif l[1] == "B"
                    total_score += 3
                elseif l[1] == "C"
                    total_score += 1
                end
            end

            println(total_score)
        end
    end
end

read_input()
